src/main/io/asyncfatfs/asyncfatfs.c

```diff
@@ -2605,8 +2607,12 @@ static void afatfs_createFileContinue(afatfsFile_t *file)
                 entry->attrib = file->attrib;
                 entry->creationDate = AFATFS_DEFAULT_FILE_DATE;
                 entry->creationTime = AFATFS_DEFAULT_FILE_TIME;
-                entry->lastWriteDate = AFATFS_DEFAULT_FILE_DATE;
-                entry->lastWriteTime = AFATFS_DEFAULT_FILE_TIME;
+                if (rtcGetDateTime(&now)) {
+                    entry->creationDate = FAT_MAKE_DATE(now.year, now.month, now.day);
+                    entry->creationTime = FAT_MAKE_TIME(now.hours, now.minutes, now.seconds);
+                }
```

Comment:
Shouldn't we move 
```
                 entry->creationDate = AFATFS_DEFAULT_FILE_DATE;
                  entry->creationTime = AFATFS_DEFAULT_FILE_TIME;
```
to an `else` clause?

Reply:
Well, I incorrectly assumed that not using the `else` would save a bit of flash since the macro should resolve to a constant a compile time. However, using the else saves 18 bytes of TEXT, so I'll amend it.